ITE/4/README.adoc

```diff
@@ -0,0 +1,283 @@
+    },
+    "...": ...
+}
+```
+
+Similarly, an SPDX URI can be used to directly fetch the corresponding documents
+from a specified location. For in-toto metadata that exist within SPDX
+documents, the URI can also be used to refer to other entities elsewhere in the
+document.
+
+Artifact Rules are a means to specify what artifacts the project owner expects
+in every step of the supply chain in the in-toto layout. in-toto povides the
+following rules that together attempt to describe the different types of
+operations or actions that supply chain functionaries carry out.
+
+=== `MATCH <pattern> [IN <source-path-prefix>] WITH (MATERIALS|PRODUCTS) [IN <destination-path-prefix>] FROM <step>`
+
+The Match rule is a convenient way to match artifacts (either in materials or
+products depending on where the rule is specified) with artifacts from other
+steps in the supply chain, allowing owners to establish a flow of artifacts
+between steps in the software supply chain.
+
+The current verification workflow for the `MATCH` rule compares the
+cryptographic hashes of the relevant artifacts from the respective link files.
+This is a straightforward comparison and does not entail resolving the URIs in
+any form. The `IN` clauses which are used to specify path prefixes are
+subtracted from the URI strings. Therefore, we conclude that the functioning of
+the rule doesn't change with this ITE. An example `MATCH` rule with generic URIs
+is as follows:
+
+`MATCH pull/250 IN github+https://github.com/in-toto/in-toto/ WITH PRODUCTS IN
+github+https://github.com/in-toto/in-toto-golang/pull/ FROM merge-go-repository`
+
+=== `ALLOW <pattern>`
+
+The Allow rule specifies an artifact matched by the rule is allowed as a
+material or product of the particular step. This rule is not affected by this
+ITE.
+
+=== `DISALLOW <pattern>`
+
+The Disallow rule specifies an artifact matched by the Rule is not allowed as a
+material or product of the particular step. This rule is not affected by this
+ITE.
+
+=== `CREATE <pattern>`
+
+The Create rule specifies that an artifact matched by this rule MUST not appear
+as a material of the particular step - it MUST instead be created as part of
+executing this step of the supply chain. This rule is not affected by this ITE.
+
+=== `DELETE <pattern>`
+
+The Delete rule specifies that an artifact matched by this rule MUST not appear
+as a product of the particular step - it MUST instead be deleted as part of
+executing this step of the supply chain. This rule is not affected by this ITE.
+
+=== `REQUIRE <pattern>`
+
+The Require rule specifies that an artifact matched by this rule MUST appear as
+a material or product of the particular step. This rule is not affected by this
+ITE.
+
+=== `MODIFY <pattern>`
+
+The Modify rule specifies that an artifact matched by this rule MUST appear as
+both a material and product of the step, and the hash of the artifact MUST
+differ indicating the execution of the step changed the artifact.
```

Comment:
Is it necessary to mention all rules here to this extent?

To me it makes sense to clarify that processing rules does not require resolving any paths, with or without this ITE. You already say this quite aptly above in the match rule section, i.e. *"...does not entail resolving the URIs in any form..."*. 

I suggest to, instead of iterating over all rules here, you only leave that one paragraph that briefly explains what artifact rules are good for, and then explain how this ITE does not affect any of the rule's functioning.

The  `"MATCH pull/250 ..."` example, on the other hand, could use some explanation about the context. What kind of supply chain are we talking about here. What does the `"merge-go-repository"` step do? Why would the exact same GitHub PR page exist on two different repos?

Reply:
> I suggest to, instead of iterating over all rules here, you only leave that one paragraph that briefly explains what artifact rules are good for, and then explain how this ITE does not affect any of the rule's functioning.

That does make sense. The MATCH rule is the only real confusing one so I think it's better to explore that a bit while just explaining the rest briefly.

> The `"MATCH pull/250 ..."` example, on the other hand, could use some explanation about the context. What kind of supply chain are we talking about here. What does the "merge-go-repository" step do? Why would the exact same GitHub PR page exist on two different repos?

That's a good point, and I'll work on an improved scenario. We're also not talking about just the pages as much as some more abstract representation of what a pull request is, but I think the point you made still stands. Thanks!